src/ledger/LedgerTxnDataSQL.cpp

```diff
@@ -127,4 +127,71 @@ LedgerTxnRoot::Impl::dropData()
                               "PRIMARY KEY  (accountid, dataname)"
                               ");";
 }
+
+static std::vector<LedgerEntry>
+loadDataToEncode(Database& db, size_t limit, size_t offset)
+{
+    std::string accountID, dataName, dataValue;
+    uint32_t lastModified;
+
+    std::string sql = "SELECT accountid, dataname, datavalue, lastmodified "
+                      "FROM accountdataunencoded "
+                      "ORDER BY accountid, dataname LIMIT :v1 OFFSET :v2";
+
+    auto prep = db.getPreparedStatement(sql);
+    auto& st = prep.statement();
+    st.exchange(soci::into(accountID));
+    st.exchange(soci::into(dataName));
+    st.exchange(soci::into(dataValue));
+    st.exchange(soci::into(lastModified));
+    st.exchange(soci::use(limit));
+    st.exchange(soci::use(offset));
+    st.define_and_bind();
+    st.execute(true);
+
+    std::vector<LedgerEntry> res;
+    while (st.got_data())
+    {
+        res.emplace_back();
+        auto& le = res.back();
+        le.data.type(DATA);
+
+        auto& de = le.data.data();
+        de.accountID = KeyUtils::fromStrKey<PublicKey>(accountID);
+        de.dataName = dataName;
+        decoder::decode_b64(dataValue, de.dataValue);
+        le.lastModifiedLedgerSeq = lastModified;
+
+        st.fetch();
+    }
+    return res;
+}
+
+void
+LedgerTxnRoot::Impl::encodeDataNamesBase64()
+{
+    throwIfChild();
+    mEntryCache.clear();
+    mBestOffersCache.clear();
+
+    mDatabase.getSession()
+        << "ALTER TABLE accountdata RENAME TO accountdataunencoded;";
+    dropData();
+
+    size_t const LIMIT = 4096;
+    size_t offset = 0;
+    auto dataToEncode = loadDataToEncode(mDatabase, LIMIT, offset);
```

Comment:
I would ditch the LIMIT offset (and loop) given that we have order of 100k data entries total

Reply:
technically if we do that we don't even need to keep the data around in `accountdataunencoded` (ie: we load all data, recreate the table, write the data), so much simpler logic ; one of the side effects of having 2 tables might be unnecessary overhead that will have to be vacuumed.

Reply:
Yeah, I agree there's unlikely to be a problem doing it all in memory.